src/components/dialogs/line-attach-to-voltage-level-dialog.js

```diff
@@ -183,9 +185,14 @@ const LineAttachToVoltageLevelDialog = ({
         });
 
     useEffect(() => {
+        const vlId =
+            typeof voltageLevelOrId === 'string'
+                ? voltageLevelOrId
+                : voltageLevelOrId?.id;
         if (
-            newVoltageLevelRef.current !== null &&
-            voltageLevelOrId !== voltageLevelOrIdRef.current
+            newVoltageLevelRef.current &&
+            voltageLevelOrIdRef.current &&
+            vlId !== newVoltageLevelRef.current.equipmentId
         ) {
             voltageLevelOrIdRef.current = voltageLevelOrId;
             setNewVoltageLevel(null);
```

Comment:
Maybe change the following line to newVoltageLevelRef.current = null;, and remove dependency to newVoltageLevel since it might be dangerous to use setNewVoltageLevel and newVoltageLevel in a useEffect.